### PR TITLE
Refs #31026 -- Fixed forms_tests if Jinja2 is not installed.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Review by carltongibson — APPROVED** [review]
Yes (Of course... :)

This does the job. Thanks @felixxm 👍

File: tests/forms_tests/tests/__init__.py [hunk 1]
```diff
@@ -19,7 +26,8 @@ def wrapper(func):
         def inner(*args, **kwargs):
             for settings in TEST_SETTINGS:
                 with override_settings(**settings):
-                    func(*args, **kwargs)
+                    test_func = skipIf(jinja2 is None, 'this test requires jinja2')(func)
```
Comment (ngnpope) [inline tests/forms_tests/tests/__init__.py:29]: This will skip Django templates tests also...

**Review by ngnpope — CHANGES_REQUESTED** [review]
I'm not sure this is right.

File: tests/forms_tests/tests/__init__.py (hunk 1 above)
Comment (carltongibson) [reply tests/forms_tests/tests/__init__.py:29]: Yes. Good. 

We need something like: 

```
                    if settings['FORM_RENDERER'] == 'django.forms.renderers.Jinja2':
```

before applying the `skipIf`. 

(It's not very pretty but...)

Comment (carltongibson) [reply tests/forms_tests/tests/__init__.py:29]: ☕

Comment (carltongibson) [reply tests/forms_tests/tests/__init__.py:29]: This won't work either. The tests run, but they're marked as skipped. @felixxm is adjusting.

**Comment by felixxm** [comment]
I switched to the existing `@require_jinja2` decorator.

**Comment by smithdc1** [comment]
> I switched to the existing `@require_jinja2` decorator.

Not at my desk but from memory I don't _think_ this is correct. IIRC it works for the widget tests as we have a custom test function that runs both engines (self.check_widget??), but we're just using the standard unit test assertions for the forms.

**Comment by felixxm** [comment]
OK, the 3rd attempt :laughing: I added the `jinja2_tests` decorator, now it works as expected.

**Comment by felixxm** [comment]
> > I switched to the existing `@require_jinja2` decorator.
> 
> Not at my desk but from memory I don't _think_ this is correct. IIRC it works for the widget tests as we have a custom test function that runs both engines (self.check_widget??), but we're just using the standard unit test assertions for the forms.

True, see https://github.com/django/django/pull/14874#issuecomment-923709987

**Review by carltongibson — APPROVED** [review]

**Pull request merged**